### DISCUSSION TITLE
Is it possible to change the PDF filename when output directly to the browser?

### DISCUSSION THREAD

**Discussion by ryanmortier** [spatie/browsershot#583] [Q&A]
If I have a route like this:

![image](https://user-images.githubusercontent.com/2053960/143984362-19fcff2a-8df3-4c12-a5bd-90b760ab85cd.png)

If the user goes to save the file after outputting directly to the browser, the default filename is the same as the route.

![image](https://user-images.githubusercontent.com/2053960/143984418-37f70de6-41c8-4010-8b7b-d08a70f32e17.png)

Is there anyway to still output directly to the browser but change the filename if the user decides to save the PDF?

**Comment by chrisbalicki** [comment]
You can set content disposition in the header and add the filename there... 

It will look something like this

```
return response()->stream(function () use ($inspection) {
            echo Browsershot::html(view('inspection-pdf', ['inspection' => $inspection])->render())
            ->margins(10, 0, 10, 0)
            ->format('A4')
            ->pdf();
        }, 200, ['Content-Type' => 'application/pdf', 
        'Content-Disposition' => 'inline; filename="this is a file name and can be variable" '
      ]);
```

This will then set the filename when saving after streaming.
